### PR TITLE
Store: standardize static city/state/postcode formatting

### PR DESCRIPTION
**Pull request by dechov**
Minor address formatting tweaks for visual appearance and text handling.

Open to feedback on the precise format itself — I've gone with how I always have written it, which is partially supported by [USPS address formatting guidelines](https://pe.usps.com/businessmail101?ViewName=DeliveryAddress) ("Two spaces between state and ZIP Code") but not entirely ("No punctuation").

- `<AddressView>` (in settings and in order customer info):

Before | After
-- | --
 | 
CSS margins | HTML string separators
_Copy & paste:_ New YorkNY10021 | _Copy & paste:_ New York, NY&nbsp; 10021

- Shipping label purchase dialog

Before | After
-- | --
 | 
 | 

Unchanged but could perhaps be discussed within the scope of this PR:
- "United States (US)" as [expanded](https://github.com/Automattic/wp-calypso/blob/b718bbe6f74ce9ba2ee2cb1738f7c44c5a800ecb/client/extensions/woocommerce/woocommerce-services/views/shipping-label/label-purchase-modal/address-step/suggestion.js#L36) in the label purchase dialog is preferable to "US"
- Not so sure about [expanding](https://github.com/Automattic/wp-calypso/blob/b718bbe6f74ce9ba2ee2cb1738f7c44c5a800ecb/client/extensions/woocommerce/woocommerce-services/views/shipping-label/label-purchase-modal/address-step/suggestion.js#L34) the State name, especially if this is lowercase while the rest of the address is being normalized to uppercase
- Not sure how best to internationalize

### CODE REVIEW

**Comment by DanReyLop** [comment]
> "United States (US)" as expanded in the label purchase dialog is preferable to "US"

+1 to that

> Not so sure about expanding the State name, especially if this is lowercase while the rest of the address is being normalized to uppercase

We should expand the state name. In US the state abbreviations are commonly used, but not so much in other countries. In Spain I'm not even sure of the abbreviation of the province where I live :)

> Not sure how best to internationalize

WooCommerce Core has some rules in the `class-wc-countries.php#get_address_formats`. We'll eventually need to copy them into Calypso or add an endpoint to retrieve them, but this is not the time for that :)

I'm OK with these changes. If you want, wait for a design review to :shipit:

**Comment by jameskoster** [comment]
Two spaces between state and zip looks a little odd to me. Definitely not something I'm used to seeing in the UK. I know we're focussing on North America to begin with but could this be localised so it's applied based on the store location or something?

All looks fine otherwise.

**Comment by allendav** [comment]
> In US the state abbreviations are commonly used, but not so much in other countries. In Spain I'm not even sure of the abbreviation of the province where I live :)

I would prefer we not try and force one way to render them all, but present the states/provinces in the format the merchants and their shipping providers will expect.

File: client/extensions/woocommerce/woocommerce-services/views/shipping-label/label-purchase-modal/address-step/index.js [hunk 1]
```diff
@@ -39,11 +39,11 @@ const renderSummary = ( {
 	}
 	const { countriesData } = storeOptions;
 	const { city, postcode, state, country } = ( normalized && selectNormalized ) ? normalized : values;
-	// Summary format: "city, state postcode [, country]"
+	// Summary format: "city, state  postcode [, country]"
 	let str = city + ', ';
 	if ( state ) {
 		const statesMap = ( countriesData[ country ] || {} ).states || {};
-		str += ( statesMap[ state ] || state ) + ' ';
+		str += ( statesMap[ state ] || state ) + '\xa0 ';
```
Comment (allendav) [inline client/extensions/woocommerce/woocommerce-services/views/shipping-label/label-purchase-modal/address-step/index.js]: Maybe add a comment explaining that `\xa0` is

**Comment by justinshreve** [comment]
Hey @dechov. I wanted to follow up on the comment above. Do we need to make any changes here?

**Comment by dechov** [comment]
@justinshreve Yes, thanks for the follow-up — I think some adjustment of abbreviations is called for — I'll get to it today or tomorrow.

**Comment by dechov** [comment]
Apologies for the delay — I wanted to expand the country name but didn't realize this was already being done elsewhere in the same component, making this a very quick change. The country line now says either "USA" or "Canada". If we want it to say "United States", we can change it in the [country data](https://github.com/Automattic/wp-calypso/blob/d3f13cb796bc3d2fa0a128279e60db8f466c4f7f/client/extensions/woocommerce/lib/countries/US.js#L17), but otherwise I believe this is ready for final review.

**Comment by allendav** [comment]
@dechov wrote

> cc @allendav for an approval since you have a red x here

wfm - :shipit:

**Pull request merged**